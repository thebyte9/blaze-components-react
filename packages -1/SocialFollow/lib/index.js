"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var react_1 = __importDefault(require("react"));
var SocialFollow = function (_a) {
    var media = _a.media, title = _a.title, vertical = _a.vertical, type = _a.type;
    var socialConf = {
        facebook: {
            "class": "facebook",
            icon: "fab fa-facebook-f",
            name: "Facebook"
        },
        instagram: {
            "class": "instagram",
            icon: "fab fa-instagram",
            isFollowing: true,
            name: "Instagram"
        },
        linkedIn: {
            "class": "linkedin",
            icon: "fab fa-linkedin-in",
            name: "Linkedin"
        },
        pinterest: {
            "class": "pinterest",
            icon: "fab fa-pinterest-p",
            name: "Pinterest"
        },
        twitter: {
            "class": "twitter",
            icon: "fab fa-twitter",
            name: "Twitter"
        },
        youtube: {
            "class": "youtube",
            icon: "fab fa-youtube",
            isFollowing: true,
            name: "Youtube"
        }
    };
    var renderSocial = Object.keys(media).map(function (key) {
        if (type !== "follow" && socialConf[key].isFollowing) {
            return null;
        }
        return (react_1["default"].createElement("li", { key: key, className: "social__list-item social__list-item--" + type },
            react_1["default"].createElement("a", { href: media[key], rel: "noopener noreferrer", target: "_blank", className: socialConf[key]["class"] },
                react_1["default"].createElement("i", { className: socialConf[key].icon }),
                react_1["default"].createElement("span", { className: "hidden" }, socialConf[key].name))));
    });
    return (react_1["default"].createElement("div", { className: "social social--" + type + " " + (vertical ? "social--vertical" : "") },
        title && react_1["default"].createElement("p", null, title),
        react_1["default"].createElement("ul", { className: "social__list social__list--" + type + " " + (vertical ? "social__list--vertical" : "") }, renderSocial)));
};
SocialFollow.defaultProps = {
    title: "",
    type: "share",
    vertical: false
};
exports["default"] = SocialFollow;
