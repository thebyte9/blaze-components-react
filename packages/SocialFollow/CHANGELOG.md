# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.



**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.30](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.29...v0.8.0-alpha.30) (2021-09-06)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.29](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.28...v0.8.0-alpha.29) (2021-07-27)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.28](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.26...v0.8.0-alpha.28) (2021-07-20)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.26](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.25...v0.8.0-alpha.26) (2021-07-20)

**Note:** Version bump only for package @blaze-react/social-follow







**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.25](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.24...v0.8.0-alpha.25) (2021-07-20)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.24](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.23...v0.8.0-alpha.24) (2021-07-20)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.23](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.22...v0.8.0-alpha.23) (2021-07-19)

**Note:** Version bump only for package @blaze-react/social-follow







**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.22](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.21...v0.8.0-alpha.22) (2021-07-19)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.21](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.20...v0.8.0-alpha.21) (2021-07-14)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.20](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.19...v0.8.0-alpha.20) (2021-07-13)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.19](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.18...v0.8.0-alpha.19) (2021-07-13)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.18](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.17...v0.8.0-alpha.18) (2021-07-13)

**Note:** Version bump only for package @blaze-react/social-follow







**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.17](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.16...v0.8.0-alpha.17) (2021-07-13)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.16](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.15...v0.8.0-alpha.16) (2021-07-13)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.15](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.14...v0.8.0-alpha.15) (2021-07-13)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.14](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.13...v0.8.0-alpha.14) (2021-07-13)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.13](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.12...v0.8.0-alpha.13) (2021-07-13)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.12](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.11...v0.8.0-alpha.12) (2021-07-13)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.11](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.10...v0.8.0-alpha.11) (2021-07-13)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.10](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.9...v0.8.0-alpha.10) (2021-07-13)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.9](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.8...v0.8.0-alpha.9) (2021-07-12)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.8](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.7...v0.8.0-alpha.8) (2021-07-12)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.7](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.6...v0.8.0-alpha.7) (2021-07-12)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.6](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.3...v0.8.0-alpha.6) (2021-07-09)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.3](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.2...v0.8.0-alpha.3) (2021-06-17)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.2](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.1...v0.8.0-alpha.2) (2021-06-17)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.1](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.0...v0.8.0-alpha.1) (2021-06-17)

**Note:** Version bump only for package @blaze-react/social-follow





# [0.8.0-alpha.0](https://github.com/thebyte9/blaze-components-react/compare/v0.6.2-alpha.0...v0.8.0-alpha.0) (2021-06-17)

**Note:** Version bump only for package @blaze-react/social-follow





## [0.6.2-alpha.0](https://github.com/thebyte9/blaze-components-react/compare/v0.6.1...v0.6.2-alpha.0) (2021-06-17)

**Note:** Version bump only for package @blaze-react/social-follow
