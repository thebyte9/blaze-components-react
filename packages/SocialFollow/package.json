{
  "name": "@blaze-react/social-follow",
  "version": "0.6.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "",
  "main": "./src/index.ts",
  "scripts": {
    "build": "NODE_ENV=production webpack --config webpack.config.js",
    "build-storybook": "build-storybook -c .storybook -o docs-build",
    "lint": "eslint '*/**/*.{js,ts,tsx}' --quiet --fix",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "pre-publish": "yarn build && yarn lint && yarn test"
  },
  "keywords": [],
  "author": "byte9",
  "license": "MIT",
  "devDependencies": {
    "@blaze-react/blaze-components-theme": "^0.6.0",
    "webpack": "^5.28.0",
    "webpack-cli": "^4.6.0"
  },
  "peerDependencies": {
    "prop-types": "^15.7.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
