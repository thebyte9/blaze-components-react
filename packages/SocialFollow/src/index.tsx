import React from "react";
type SocialFollowProps = {
  type?: string,
  title?: string,
  vertical?: boolean,
  media: object
};
const SocialFollow: React.SFC<SocialFollowProps> = ({
  media,
  title,
  vertical,
  type
}) => {
  const socialConf = {
    facebook: {
      name: "Facebook",
      class: "facebook",
      icon: "fab fa-facebook-f"
    },
    twitter: {
      name: "Twitter",
      class: "twitter",
      icon: "fab fa-twitter"
    },
    pinterest: {
      name: "Pinterest",
      class: "pinterest",
      icon: "fab fa-pinterest-p"
    },
    linkedIn: {
      name: "Linkedin",
      class: "linkedin",
      icon: "fab fa-linkedin-in"
    },
    youtube: {
      name: "Youtube",
      class: "youtube",
      icon: "fab fa-youtube",
      isFollowing: true
    },
    instagram: {
      name: "Instagram",
      class: "instagram",
      icon: "fab fa-instagram",
      isFollowing: true
    }
  };
  const renderSocial = Object.keys(media).map(key => {
    if (type !== "follow" && socialConf[key].isFollowing) return null;
    return (
      <li key={key} className={`social__list-item social__list-item--${type}`}>
        <a
          href={media[key]}
          rel="noopener noreferrer"
          target="_blank"
          className={socialConf[key].class}
        >
          <i className={socialConf[key].icon} />
          <span className="hidden">{socialConf[key].name}</span>
        </a>
      </li>
    );
  });
  return (
    <div
      className={`social social--${type} ${vertical ? "social--vertical" : ""}`}
    >
      {title && <p>{title}</p>}
      <ul
        className={`social__list social__list--${type} ${
          vertical ? "social__list--vertical" : ""
        }`}
      >
        {renderSocial}
      </ul>
    </div>
  );
};
SocialFollow.defaultProps = {
  type: "share",
  title: "",
  vertical: false
};
export default SocialFollow;
