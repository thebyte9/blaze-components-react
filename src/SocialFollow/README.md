
## Description

The Social Icons Widget displays small graphics linked to your social media accounts,
and can be displayed on your site in different ways.

## Usage

* Share

```js
const media = {
    facebook: 'https://www.thebyte9.com',
    twitter: 'https://www.thebyte9.com',
    pinterest: 'https://www.thebyte9.com',
    linkedIn: 'https://www.thebyte9.com',
    youtube: 'https://www.thebyte9.com',
    instagram: 'https://www.thebyte9.com'
};

<SocialFollow media={media} vertical/> 
```

* Follow

```js
<SocialFollow media={media} type="follow"/> 
```

## API

##### SocialFollow can receive a number of `props` as follow:


| NAME   | TYPE | DEFAULT | 
| :---  | :---:  | :---: | 
| title | String | empty | 
| type | String | share | 
| media | Object | {} |
| vertical | Boolean | false |




