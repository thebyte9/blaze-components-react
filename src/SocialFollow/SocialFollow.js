import React from 'react';
import PropTypes from 'prop-types';

const SocialFollow = ({
  media: {
    facebook,
    twitter,
    pinterest,
    LinkedIn,
    youtube,
    instagram
  },
  type,
  title,
  vertical
}) => (
  <div className={`social social--${type} ${vertical ? 'social--vertical' : ''}`}>
    <p>{title || type.charAt(0).toUpperCase() + type.slice(1)}</p>
    <ul className={`social__list social__list--${type} ${vertical ? 'social__list--vertical' : ''}`}>
      {facebook && (
        <li className={`social__list-item social__list-item--${type}`}>
          <a href={facebook} rel="noopener noreferrer" target="_blank" className="facebook">
            <i className="fab fa-facebook-f" />
            <span className="hidden">Facebook</span>
          </a>
        </li>
      )}
      {twitter && (
        <li className={`social__list-item social__list-item--${type}`}>
          <a href={twitter} rel="noopener noreferrer" target="_blank" className="twitter">
            <i className="fab fa-twitter" />
            <span className="hidden">Twitter</span>
          </a>
        </li>
      )}
      {pinterest && (
      <li className={`social__list-item social__list-item--${type}`}>
        <a href={pinterest} rel="noopener noreferrer" target="_blank" className="pinterest">
          <i className="fab fa-pinterest-p" />
          <span className="hidden">Pinterest</span>
        </a>
      </li>
      )}
      {LinkedIn && (
      <li className={`social__list-item social__list-item--${type}`}>
        <a href={LinkedIn} rel="noopener noreferrer" target="_blank" className="linkedin">
          <i className="fab fa-linkedin-in" />
          <span className="hidden">LinkedIn</span>
        </a>
      </li>
      )}
      {youtube && type === 'follow' && (
      <li className={`social__list-item social__list-item--${type}`}>
        <a href={youtube} rel="noopener noreferrer" target="_blank" className="youtube">
          <i className="fab fa-youtube" />
          <span className="hidden">YouTube</span>
        </a>
      </li>
      )}
      {instagram && type === 'follow' && (
      <li className={`social__list-item social__list-item--${type}`}>
        <a href={instagram} rel="noopener noreferrer" target="_blank" className="instagram">
          <i className="fab fa-instagram" />
          <span className="hidden">Instagram</span>
        </a>
      </li>
      )}
    </ul>
  </div>
);

SocialFollow.propTypes = {
  media: PropTypes.object.isRequired,
  type: PropTypes.string,
  title: PropTypes.string,
  vertical: PropTypes.bool
};

SocialFollow.defaultProps = {
  type: 'share',
  title: '',
  vertical: false
};

export default SocialFollow;
