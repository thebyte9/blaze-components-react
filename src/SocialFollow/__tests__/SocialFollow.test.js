import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import SocialFollow from '../SocialFollow';

const media = {
  facebook: 'https://www.thebyte9.com',
  twitter: 'https://www.thebyte9.com',
  pinterest: 'https://www.thebyte9.com',
  LinkedIn: 'https://www.thebyte9.com',
  youtube: 'https://www.thebyte9.com',
  instagram: 'https://www.thebyte9.com'
};

describe('SocialFollow component', () => {
  test('should be defined and renders correctly (snapshot)', () => {
    const wrapper = shallow(<SocialFollow media={media} type="follow" vertical />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').at(0)).toContain('Follow');
  });
  test('SocialFollow should be share by default', () => {
    const wrapper = shallow(<SocialFollow media={media} />);
    expect(wrapper.find('.social--share').length).toEqual(1);
  });
});
