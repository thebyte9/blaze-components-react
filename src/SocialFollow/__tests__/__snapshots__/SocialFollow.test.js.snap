// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialFollow component should be defined and renders correctly (snapshot) 1`] = `ShallowWrapper {}`;
